src/compiler/binder.ts

```diff
@@ -992,6 +1128,14 @@ namespace ts {
 
             let symbol = node.symbol;
 
+            // For a class with no declared properties in a JS file, we can infer its members from assignments
+            // of the form 'this.name = expr' in the constructor. Set the flag that indicates
+            // when that should happen.
+            if (isJavaScriptFile) {
+                let hasDeclaredProperties = forEachChild(node, child => child.kind === SyntaxKind.PropertyDeclaration);

```

Comment:
We might be offering intellisense for a Babel- or Flow-targeted JS file, so I think anything is really up for grabs in terms of supported syntax. But I don't have a strong intuition either way on what our behavior ought to be.